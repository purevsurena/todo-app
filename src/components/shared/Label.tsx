import * as React from "react";
import { StyleSheet, ViewStyle } from "react-native";
import Click from "@/components/layouts/Click";
import { ThemedText } from "@/components/shared/ThemedText";
import { ThemedView } from "@/components/shared/ThemedView";
import { Colors } from "@/constants/Colors";

interface LabelProps {
  onPress: () => void;
  active: boolean;
  label: string;
}

const Label: React.FC<LabelProps> = ({ onPress, active, label }) => {
  return (
    <Click onPress={onPress}>
      <ThemedView
        style={[
          styles.labelContainer,
          {
            borderColor: active
              ? Colors.generic.turquoise
              : Colors.generic.label,
          },
        ]}
      >
        <ThemedText>{label}</ThemedText>
      </ThemedView>
    </Click>
  );
};

export default Label;

const styles = StyleSheet.create({
  labelContainer: {
    paddingHorizontal: 12,
    paddingVertical: 4,
    borderRadius: 12,
    backgroundColor: Colors.generic.label,
    marginLeft: 8,
    marginBottom: 8,
    borderWidth: 2,
  } as ViewStyle,
});
