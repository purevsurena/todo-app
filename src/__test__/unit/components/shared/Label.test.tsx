import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import Label from "@/components/shared/Label"; // Adjust the import path as necessary

describe("Label Component", () => {
  const onPressMock = jest.fn();
  const label = "Test Label";

  it("renders correctly with inactive state", () => {
    const { getByText } = render(
      <Label onPress={onPressMock} active={false} label={label} />,
    );

    const labelText = getByText(label);

    expect(labelText).toBeTruthy();
  });

  it("calls onPress when clicked", () => {
    const { getByText } = render(
      <Label onPress={onPressMock} active={false} label={label} />,
    );

    const labelText = getByText(label);
    fireEvent.press(labelText);

    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it("matches the snapshot", () => {
    const { toJSON } = render(
      <Label onPress={onPressMock} active={false} label={label} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
