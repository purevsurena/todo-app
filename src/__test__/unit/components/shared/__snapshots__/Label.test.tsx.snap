// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Component matches the snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      [
        {
          "backgroundColor": "#3C3C3C",
          "borderRadius": 12,
          "borderWidth": 2,
          "marginBottom": 8,
          "marginLeft": 8,
          "paddingHorizontal": 12,
          "paddingVertical": 4,
        },
        {
          "borderColor": "#3C3C3C",
        },
      ],
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#fff",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ]
    }
  >
    Test Label
  </Text>
</View>
`;
